### PR TITLE
Stream subscription delay in LedgerApiBenchTool [DPP-1209]

### PR DESCRIPTION
**Pull request by tudor-da**
changelog_begin
changelog_end

### CODE REVIEW

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/Benchmark.scala [hunk 1]
```diff
@@ -41,87 +35,92 @@ object Benchmark {
...
-              itemCountingFunction = MetricsSet.countFlatTransactionsEvents,
-              maxItemCount = streamConfig.maxItemCount,
-            )(system, ec)
-            .flatMap { observer =>
-              streamConfig.timeoutInSecondsO
-                .foreach(timeout => scheduleCancelStreamTask(timeout, observer))
-              apiServices.transactionService.transactions(streamConfig, observer)
-            }
+          for {
+            _ <- delaySubscriptionIfConfigured(streamConfig)(system)
```
Comment (pbatko-da) [inline ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/Benchmark.scala:39]: You're adding the delay in a few places here. Could you instead add it once in `com.daml.ledger.api.benchtool.LedgerApiBenchTool#benchmarkStreams` which is the only call site of `Benchmark.run`?

Comment (pbatko-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/Benchmark.scala:39]: Oh, you're delaying each stream separately which is more powerful and straightforward to do

File: ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala [hunk 2]
```diff
@@ -125,6 +129,7 @@ class WorkflowConfigParserSpec extends AnyWordSpec with Matchers {
               ),
               maxItemCount = Some(700),
               timeoutInSecondsO = None,
+              subscriptionDelay = Some(Duration(7, TimeUnit.MINUTES)),
```
Comment (pbatko-da) [inline ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala:132]: Is there a test to verify this configuration key is optional? `subscriptionDelay = None`

**Review by pbatko-da — APPROVED** [review]

File: ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala (hunk 2 above)
Comment (tudor-da) [reply ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala:138]: added 👍

**Pull request merged**